numpy/linalg/linalg.py

```diff
@@ -596,47 +585,56 @@ def matrix_power(M, n):
            [ 1.,  0.,  0.,  0.],
            [ 0.,  0.,  0.,  1.],
            [ 0.,  0., -1.,  0.]])
-    >>> LA.matrix_power(q, 2) # = -np.eye(4)
+    >>> matrix_power(q, 2) # = -np.eye(4)
     array([[-1.,  0.,  0.,  0.],
            [ 0., -1.,  0.,  0.],
            [ 0.,  0., -1.,  0.],
            [ 0.,  0.,  0., -1.]])
 
     """
-    M = asanyarray(M)
-    if M.ndim != 2 or M.shape[0] != M.shape[1]:
+    a = array(a, subok=True, copy=False, ndmin=1)
+    m = a.shape[-1]
+    if a.ndim < 2 or a.shape[-2] != m:
         raise ValueError("input must be a square array")
```

Comment:
I think we have some helper functions for this kind of check in this file. `_assert2DSquareness` or similar